lib/zold/node/safe_entrance.rb

```diff
@@ -60,16 +61,19 @@ def push(id, body)
         File.write(f, body)
         wallet = Wallet.new(f.path)
         unless wallet.network == @network
-          raise "The network name mismatch, the wallet is in '#{wallet.network}', we are in '#{@network}'"
+          raise Error::InvalidWallet,
+                "The network name mismatch, the wallet is in '#{wallet.network}', we are in '#{@network}'"
         end
         balance = wallet.balance
         if balance.negative? && !wallet.root?
-          raise "The balance #{balance} of #{wallet.id} is negative and it's not a root wallet"
+          raise Error::NegativeBalance,
+                "The balance #{balance} of #{wallet.id} is negative and it's not a root wallet"
         end
         Emission.new(wallet).check
         tax = Tax.new(wallet)
         if tax.in_debt?
-          raise "Taxes are not paid, can't accept the wallet; the debt is #{tax.debt} (#{tax.debt.to_i} zents)"
+          raise Error::NegativeBalance,
```

Comment:
@jedediah it looks like you're using the same exception for different errors. As far as I know, tax debt doesn't automatically mean negative balance.